src/coord/src/catalog/storage.rs

```diff
@@ -244,6 +244,39 @@ impl Connection {
         res
     }
 
+    pub fn get_catalog_content_version(&mut self) -> Result<usize, Error> {
+        let tx = self.inner.transaction()?;
+        let current_setting: Option<String> = tx
+            .query_row(
+                "SELECT value FROM settings WHERE name = 'catalog_content_version';",
+                params![],
+                |row| row.get(0),
+            )
+            .optional()?;
+        let version = match current_setting {
+            Some(v) => v.parse::<i64>().unwrap() as usize,
+            None => {
+                tx.execute(
+                    "INSERT INTO settings (name, value) VALUES ('catalog_content_version', 0);",
+                    params![],
+                )?;
+                0
+            }
+        };
+        tx.commit()?;
+        Ok(version as usize)
+    }
+
+    pub fn set_catalog_content_version(&mut self, new_version: usize) -> Result<(), Error> {
+        let tx = self.inner.transaction()?;
+        tx.execute(
+            "UPDATE settings SET value = ? WHERE name = 'catalog_content_version'",
+            params![new_version as i64],
```

Comment:
Casting like this is sketchy. (Yes, we'll have other problems if we have more than `i64::MAX` migrations, but it still stresses me out!) `u32::try_from(new_version)` is probably the way to go—`u32` for parity with the recommendation above.